src/provider/provider_os_memory.c

```diff
@@ -387,6 +450,15 @@ static int os_mmap_aligned(void *hint_addr, size_t length, size_t alignment,
     return 0;
 }
 
+static int get_membind(os_memory_provider_t *provider, size_t size) {
+    if (provider->nodeset_len == 1) {
+        return 0;
+    }
+
+    size_t s = util_fetch_and_add64(&provider->alloc_size, size);
+
+    return (s / provider->part_size) % provider->nodeset_len;
```

Comment:
what if size is > part_size?

Reply:
I known that i forgot about something important... I added missing loop in os_allocate function

Reply:
please add an assert here

Reply:
for ```provider->part_size!= 0``` right? TBH it doen't matter - you will get an exception and crash anyway 🤣

Reply:
I think coverity will need if + return